index.d.ts

```diff
@@ -0,0 +1,133 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction<S> {
+    (error: S): void
+  }
+
+  interface FunctionReturning<T> {
+    (...t: any[]): T
+  }
+
+  interface Cancel {
+    (): void
+  }
+
+  export abstract class IFuture<L, R> {
+    fork (reject: RejectFunction<L>, resolve: ResolveFunction<R>): Cancel
+  }
+
+  export function after<L, R> (duration: number, value: string): IFuture<L, R>
+
+  export function and<L, R> (a: IFuture<L, R>, b: IFuture<L, R>): IFuture<L, R>
```

Comment:
We can relax this signature a little bit by using allowing `a` to be a `IFuture<L, any>`, since the `R` of `a` is always ignored. Let's also rename `a` to `left` and `b` to `right`.